randovania/gui/widgets/game_validator_widget.py

```diff
@@ -200,61 +222,96 @@ def write_to_log(*a: Any) -> None:
             while self._current_tree[-1].indent >= indent:
                 self._current_tree.pop().item.setExpanded(False)
 
-            if indent == 0:
-                parent = self.log_widget
-            else:
-                parent = self._current_tree[-1].item
+            item = QtWidgets.QTreeWidgetItem()
+            region, area, node = ("", "", "")
+
+            def action_type_and_text(_action: str) -> tuple[str, str]:
+                action_type_match = action_type_re.match(_action)
+                if action_type_match is None:
+                    return "Other", _action
+
+                action_text = action_type_match.group("action")
+                action_type = action_type_match.group("type")
+
+                if (action_type in {"Major", "Minor", "Pickup"}) and action_text != "Nothing":
+                    pickup_match = pickup_action_re.match(action_text)
+                    assert pickup_match is not None
 
-            item = QtWidgets.QTreeWidgetItem(parent)
+                    player = int(pickup_match.group("world_num"))
+                    pickup = pickup_match.group("pickup_name")
+
+                    if self.layout_description.generator_parameters.world_count == 1:
```

Comment:
Could be just `self.layout_description.world_count`

Reply:
intentionally future-proofed for multiworld

Reply:
Sure, but that's not what I said :P
`layout_description.world_count` is an alias for `layout_description.generator_parameters.world_count`

Reply:
oh, sure. I thought you meant the whole condition could just be that